Userland/Libraries/LibGfx/ImageFormats/TIFFLoader.cpp

```diff
@@ -263,23 +290,33 @@ class TIFFLoadingContext {
                 Optional<Color> last_color {};
 
                 for (u32 column = 0; column < *m_metadata.image_width(); ++column) {
-                    auto color = TRY(read_color(*decoded_stream));
-
-                    if (m_metadata.predictor() == Predictor::HorizontalDifferencing && last_color.has_value()) {
-                        color.set_red(last_color->red() + color.red());
-                        color.set_green(last_color->green() + color.green());
-                        color.set_blue(last_color->blue() + color.blue());
+                    if (metadata().photometric_interpretation() == PhotometricInterpretation::CMYK) {
+                        auto const cmyk = TRY(read_color_cmyk(*decoded_stream));
+                        oriented_bitmap.get<ExifOrientedCMYKBitmap>().set_pixel(column, scanline, cmyk);
+                    } else {
+                        auto color = TRY(read_color(*decoded_stream));
+
+                        // FIXME:  We should do the differencing at the byte-stream level, that would make it
+                        //         compatible with both LibPDF and all color formats.
```

Comment:
I suppose that's a good reason for adding `!= 8` bpp support to the LibPDF code and then moving it over. LMK if you'd like me to take a stab at that.

Reply:
If you want to have that soon, go ahead. I'm fine with either of us implementing the feature. Let's just warn the other before doing it :)